js/src/lib/components/InferenceWidget/shared/types.ts

```diff
@@ -66,3 +66,20 @@ export interface ImageSegment {
 	imgData?: ImageData;
 	bitmap?:  ImageBitmap;
 }
+
+export interface WidgetInput {
+	// example functions
+	validateExample:   <TOutput>(sample: WidgetExample<TOutput>) => sample is WidgetExample<TOutput>;
+	applyExample:      (sample: any, opts: ExampleRunOpts) => void;
+	// ingerence api functions
+	getInferenceInput: () => Promise<any>;
+}
+
+export interface WidgetOutput {
```

Comment:
here, i tried to explicitly differentiate between `validateExample` and `showInferenceOutput`. On most cases, these two functions should be identical. However, there are instances where it can differ, [isValidOutput](https://github.com/huggingface/hub-docs/blob/6cc444ef92c35788bdf24791bb2ee4f30d35783e/js/src/lib/components/InferenceWidget/widgets/AudioToAudioWidget/AudioToAudioWidget.svelte#L115) is different from [isValidOutputUrl](https://github.com/huggingface/hub-docs/blob/b2ba06d025a6de85049e6470f1533f6f59b4e9e8/js/src/lib/components/InferenceWidget/shared/outputValidation.ts#L20)